### PR TITLE
Backport std::make_unique

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

File: core/foundation/inc/ROOT/RMakeUnique.hxx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace std {
+
+template <typename T, typename... Args>
+std::unique_ptr<T> make_unique(Args &&... args)
+{
+   return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
+}
+}
+#endif
```
Comment (pcanal) [inline core/foundation/inc/ROOT/RMakeUnique.hxx]: For consistency, shouldn't there be a 
```
#else
// include the official one
#include <memory> 
#endif
```

**Comment by pcanal** [comment]
I am guessing that the switch from ```pragma class``` to ```pragma namespace``` prevents the autoparsing of TClassEdit.

**Comment by dpiparo** [comment]
TClassEdit was selected as class which does not have any effect. A more radical approach would be to remove this completely since the header is in the pch and by definition libCore always loaded.

File: core/foundation/inc/ROOT/RMakeUnique.hxx (hunk 1 above)
Comment (dpiparo) [reply core/foundation/inc/ROOT/RMakeUnique.hxx]: fair enough.

**Comment by pcanal** [comment]
> TClassEdit was selected as class which does not have any effect. 

humm ... so I don't understand why the test works without this PR but fail with this PR ... I am confused ...

File: core/foundation/CMakeLists.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ROOT_GLOB_SOURCES(sources ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cxx)
 
 ROOT_GLOB_HEADERS(headers ${CMAKE_CURRENT_SOURCE_DIR}/inc/*.h)
+ROOT_GLOB_HEADERS(headers ${CMAKE_CURRENT_SOURCE_DIR}/inc/ROOT/*.hxx)
```
Comment (dpiparo) [inline core/foundation/CMakeLists.txt]: This is  a mistake. The other headers were removed.

**Comment by dpiparo** [comment]
The tests fail because of the mistake above. I am about to make this right and add everything to the PCH, including TClassEdit (as it was done before).

**Comment by dpiparo** [comment]
Hi @pcanal , all comments addressed and mistake fixed.

**Pull request merged**